### ISSUE TITLE
Installation problem on Windows

### ISSUE THREAD

**Issue by tianyi2013** [ethereum/pyethereum#850]
Hi guys, I am having problem running the installation. I am on Windows 7, python 3.5.2 and Visual Studio 2015. When I run python setup.py install, it is complaining `src/python/core.c(2): fatal error C1083: Cannot open include file: 'alloca.h': No such file or directory`. I notice this is happening when it tries to install ethash. Looking at an old issue posted here I also notice there was also some blockers around LevelDB (https://github.com/ethereum/pyethereum/issues/167). I am wondering is pyethereum ready on Windows yet?

**Comment by CryptoRootz** [comment]
yea im dealing with the same problem for python 3.6 and VS 2017 ......

**Comment by fireflystorm** [comment]
Same here.
Any development on how to install it on Windows?

**Comment by robertmagier** [comment]
Yep. I am having exactly same problem.

**Comment by fireflystorm** [comment]
So this is how I went around this problem.
If you look at the error message you will find out that it is related to a failure installing ethash(aka pyethash) library and not ethereum(aka pyethereum) itself.
If you look at ethash's github issue tracker you'll see quite a few issues related to installing it on Windows and will realise that basically Windows is not supported(!)...
That said someone came up with a hack that indeed allows you to install pyethash - https://github.com/ethereum/ethash/issues/104. The hack requires you to clone the source code from github and modify two files.
You can then install ethash and after that ethereum itself.
Hope this helps.

**Comment by robertmagier** [comment]
Yes. I saw it. For me the reason to install was to be able to sign ethereum
transactions. Finally I installed web3.py beta version to sign it and
ignorres pyethereum all together. If I have a problem with I will do as you
did. Thank you.
